object/user_provider.go

```diff
@@ -0,0 +1,94 @@
+package object
+
+import (
+	"context"
+)
+
+type UserProvider struct {
+	ProviderDisplayName string `xorm:"-" json:"providerDisplayName"`
+
+	CreatedTime     string `xorm:"varchar(100)" json:"createdTime"`
+	LastSignInTime  string `xorm:"varchar(100)" json:"lastSignInTime"`
+	UsernameFromIdp string `xorm:"varchar(100)" json:"usernameFromIdp"`
+	ProviderName    string `xorm:"varchar(100)" json:"providerName"`
+	UserId          string `xorm:"varchar(100)" json:"userId"`
+	Owner           string `xorm:"varchar(100)" json:"owner"`
+}
+
+func GetGlobalUserProviders() ([]*UserProvider, error) {
+	var userProviders []*UserProvider
+
+	err := ormer.Engine.Asc("last_sign_in_time").Find(&userProviders, &UserProvider{})
+	if err != nil {
+		return userProviders, err
+	}
+
+	for _, userProvider := range userProviders {
+		err := fillProviderDisplayName(userProvider)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return userProviders, nil
+}
+
+func GetUserProviders(owner string) ([]*UserProvider, error) {
+	var userProviders []*UserProvider
+
+	err := ormer.Engine.Where("owner = ? or owner = ?", "admin", owner).Asc("last_sign_in_time").Find(&userProviders, &UserProvider{})
+	if err != nil {
+		return userProviders, err
+	}
+
+	for _, userProvider := range userProviders {
+		err := fillProviderDisplayName(userProvider)
+		if err != nil {
+			return nil, err
```

Comment:
I suggest to using a single code style
`return nil, err` or `return userProviders, err` everywhere in this func body